src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -888,6 +888,15 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
 
                 HttpResponseMessage responseMessage = WinHttpResponseParser.CreateResponseMessage(state, _doManualDecompressionCheck);
                 state.Tcs.TrySetResult(responseMessage);
+
+                // HttpStatusCode cast is needed for 308 Moved Permenantly, which we support but is not included in NetStandard status codes.
+                if (((responseMessage.StatusCode >= HttpStatusCode.MultipleChoices && responseMessage.StatusCode <= HttpStatusCode.SeeOther) ||
+                     (responseMessage.StatusCode >= HttpStatusCode.RedirectKeepVerb && responseMessage.StatusCode <= (HttpStatusCode)308)) &&
+                    state.RequestMessage.RequestUri.Scheme == Uri.UriSchemeHttps && responseMessage.Headers.Location?.Scheme == Uri.UriSchemeHttp)
+                {
+                    WinHttpTraceHelper.Trace("WinHttpHandler.SendAsync: Insecure https to http redirect from" +
+                        $"{state.RequestMessage.RequestUri} to {responseMessage.Headers.Location} blocked.");
```

Comment:
See my comment on the CurlHandler change.  There are similar costs here.  More, actually, as this will create an extra string for concatenating the const string with the result of the string interpolation.

Reply:
This should be mostly resolved. I can save the cost of the call to format, but do still incur the cost of casting both URIs to strings. I could fix that, but as far as I can tell I would have to modify WinHttpTraceHelper to do that.